packages/typescript/internal/ts_project.bzl

```diff
@@ -333,6 +361,48 @@ def _out_paths(srcs, outdir, rootdir, allow_js, ext_map):
         if _is_ts_src(f, allow_js)
     ]
 
+def _calculate_js_outs(srcs, out_dir, root_dir, allow_js, preserve_jsx, emit_declaration_only):
+    if emit_declaration_only:
+        return []
+
+    exts = {
+        "*": ".js",
+        ".jsx": ".jsx",
+        ".tsx": ".jsx",
+    } if preserve_jsx else {"*": ".js"}
+    return _out_paths(srcs, out_dir, root_dir, allow_js, exts)
+
+def _calculate_map_outs(srcs, out_dir, root_dir, source_map, preserve_jsx, emit_declaration_only):
+    if not source_map or emit_declaration_only:
+        return []
+
+    exts = {
+        "*": ".js.map",
+        ".tsx": ".jsx.map",
+    } if preserve_jsx else {"*": ".js.map"}
+    return _out_paths(srcs, out_dir, root_dir, False, exts)
+
+def _calculate_typings_outs(srcs, typings_out_dir, root_dir, declaration, composite, allow_js):
+    typings_out = []
+    #    typings_out = [
```

Comment:
comment could be removed?

Reply:
Yeah, I am still working on getting all of the tests to pass. This is remnants from that.